### ISSUE TITLE
prices should always show market prices with all decimal places

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#1107]
A confusing bug that bit me several times: if you had a `commodity` directive limiting a commodity's precision, say to two decimal places, that would also affect the output of the `prices` command. Now `prices` shows market prices (from P directives, --costs, and --inverted-costs) 1. with all decimal places and 2. otherwise with the appropriate commodity style.

**Comment by psb777** [comment]
This issue is marked "closed" and so therefore seems to be rejected. But the bug still exists (in version 1.14 at least)
If I set a price thusly
P 2020/01/16 ZAR 0.053317
then this price is correctly used in hledger and an amount 1000 ZAR is shown with -V value 53.32
However the prices report incorrectly shows the price like this
P 2020/1/16 ZAR 0.05
The commodity format is used but it should not be! This is a price, and all digits are significant.

**Comment by simonmichael** [comment]
Hi @psb777, I can't reproduce this with latest or 1.14. Could you show minimal reproducible example.

**Comment by psb777** [comment]
Sorry @simonmichael, this shortest of examples makes the point rather well :-)

```
$ cat x.ledger              
commodity 1,000,000.
P 2020/01/16 XRP 0.17474
$ 
$ hledger -f x.ledger prices
P 2020-01-16 XRP 0
```

**Comment by simonmichael** [comment]
Ah there it is, thanks. Indeed the bug was still in 1.14, it was fixed by 1.16: https://hledger.org/release-notes.html#hledger-cli-1-16.
